mne/epochs.py

```diff
@@ -3367,17 +3356,27 @@ def _concatenate_epochs(epochs_list, with_data=True, add_offset=True, *,
         evs = epochs.events.copy()
         # add offset
         if add_offset:
-            evs[:, 0] += events_offset
-        # Update offset for the next iteration.
-        events_offset = _update_offset(events_offset, epochs.events, shift)
+            if len(evs) == 0:
+                warn('One of the Epochs objects to concatenate was empty.')
+            else:
```

Comment:
We probably always want to warn, not just when `add_offset == True`, so probably this is better structured as:
```
if len(evs) == 0:
    warn(...)
elif add_offset:
    ...
```